src/collections/map.rs

```diff
@@ -340,7 +420,9 @@ where
                     };
 
                     let decoded_backing_key: K = Decode::decode(backing_key.as_slice())?;
-                    let key_cmp = map_key.cmp(&decoded_backing_key);
+
+                    //so compare backing_key with map_key.inner_bytes
+                    let key_cmp = map_key.inner.cmp(&decoded_backing_key);
```

Comment:
Shouldn't this be comparing the encoded bytes with the backing key, rather than the key values?

Reply:
Updated this. I want to revisit the logic here at some point for superfluous decodes and encodes, but will open a separate PR for this.